src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -0,0 +1,360 @@
+{
+    if (null === $this->PROPERTY) {
+        $this->PROPERTY = new CLASS($value);
+    } elseif ([] !== $value) {
+        throw new \Symfony\Component\Config\Definition\Exception\InvalidConfigurationException(sprintf(\'The node created by "NAME()" has already been initialized. You cannot pass values the second time you call NAME().\'));
+    }
+
+    return $this->PROPERTY;
+}';
+        $class->addMethod($node->getName(), $body, ['PROPERTY' => $property->getName(), 'CLASS' => $childClass->getFqcn()]);
+
+        $this->buildNode($node, $childClass, $namespace.'\\'.$childClass->getName());
+    }
+
+    private function handleVariableNode(VariableNode $node, ClassBuilder $class)
+    {
+        $comment = $this->getComment($node);
+        $property = $class->addProperty($node->getName());
+
+        $body = '
+/**
+COMMENT */
+public function NAME($value): self
+{
+    $this->PROPERTY = $value;
+
+    return $this;
+}';
+        $class->addMethod($node->getName(), $body, ['PROPERTY' => $property->getName(),  'COMMENT' => $comment]);
+    }
+
+    private function handlePrototypedArrayNode(PrototypedArrayNode $node, ClassBuilder $class, string $namespace)
+    {
+        $name = $this->getSingularName($node);
+        $prototype = $node->getPrototype();
+        $methodName = $name;
+
+        $parameterType = $this->getParameterType($prototype);
+        if (null !== $parameterType || $prototype instanceof ScalarNode) {
+            $property = $class->addProperty($node->getName());
+            if (null === $key = $node->getKeyAttribute()) {
+                $body = '
+public function NAME(TYPE $value): self
+{
+    $this->PROPERTY = $value;
+
+    return $this;
+}';
+                $class->addMethod($methodName, $body, ['PROPERTY' => $property->getName(), 'TYPE' => $parameterType]);
+            } else {
+                $body = '
+public function NAME(string $VAR, TYPE $VALUE): self
+{
+    $this->PROPERTY[$VAR] = $VALUE;
+
+    return $this;
+}';
+
+                $class->addMethod($methodName, $body, ['PROPERTY' => $property->getName(), 'TYPE' => $parameterType, 'VAR' => '' === $key ? 'key' : $key, 'VALUE' => 'value' === $key ? 'data' : 'value']);
+            }
+
+            return;
+        }
+
+        $childClass = new ClassBuilder($namespace, $name);
+        $class->addRequire($childClass);
+        $this->classes[] = $childClass;
+        $property = $class->addProperty($node->getName(), $childClass->getName().'[]');
+
+        if (null === $key = $node->getKeyAttribute()) {
+            $body = '
+public function NAME(array $value = []): CLASS
+{
+    return $this->PROPERTY[] = new CLASS($value);
+}';
+            $class->addMethod($methodName, $body, ['PROPERTY' => $property->getName(), 'CLASS' => $childClass->getFqcn()]);
+        } else {
+            $body = '
+public function NAME(string $VAR, array $VALUE = []): CLASS
+{
+    return $this->PROPERTY[$VAR] = new CLASS($VALUE);
```

Comment:
I suggest to return the object if it has already be initialized previously.
ie.
```
$security->firewall('main')->lazy(true);
$security->firewall('main')->provider('users_in_memory');
```

```suggestion
    return $this->PROPERTY[$VAR] ?? $this->PROPERTY[$VAR] = new CLASS($VALUE);
```

Reply:
Good idea!